### ISSUE TITLE
[Bug]: [Experiment] [Crash] The app crashes when navigating back to the settings menu after disabling studies

### ISSUE THREAD

**Issue by SoftVision-CiprianMuresan** [mozilla-mobile/fenix#20919]
### Steps to reproduce

**[Prerequisites]:**
- Have a local Beta build that is set-up to point to the Nimbus staging server: [Link](https://drive.google.com/file/d/1W-HXqetBe_fUojSyRDWpmvKouU9vYUJa/view?usp=sharing)
- Have the local build installed on a device.

**[Steps to reproduce]:**
1. Open the local build from the prerequisites.
2. Navigate to the Firefox settings menu.
3. Scroll down to the bottom of the list and tap the "About Firefox Beta" option.
4. Tap the "Firefox Browser" logo 5 times and go back to the settings menu.
5. Scroll down to the bottom of the list and tap the "Nimbus Experiments" option.
6. Tap the "[ppop][1166] Default Holdback - Beta" option.
7. Tap the "default_browser_settings_menu" option.
8. Navigate away from the settings menu and return.
9. Navigate to the "Data Collection" menu.
10. Tap the "Studies" option and tap the left arrow from the top right part of the browser.
11. Observe the behavior.

### Expected behaviour

- The "Set as default" banner is no longer displayed in the settings menu.

### Actual behaviour

- The app is forcibly closed and a crash notification is displayed.

### Device name

See Additional information for the list of devices we've confirmed the crash on.

### Android version

Android 8, 9, and 10

### Firefox release type

Firefox Beta

### Firefox version

92

### Device logs

https://crash-stats.mozilla.org/report/index/c83d7300-76ba-4b1b-8956-a6ac00210819

### Additional information

Reproducible on the following devices:
- Samsung Galaxy S8, Android 9 (One UI 1.0); 
- OnePlus 6, Android 10, OxygenOS 10.3.12; 
- Galaxy S9 - Android 8; 
- OnePlus 6T - Android 10,  
- OxygenOS 10.3.12; 
- HUAWEI P20 lite, Android 8.0.0, EMUI 8.0.0

- The crash is nor reproducible on the other branch of the experiment.
- The issue was also reproducible on a Firefox 91 Release local build.
- [Link to the Nimbus Experiment on the Staging server](https://stage.experimenter.nonprod.dataops.mozgcp.net/nimbus/ppop-1166-default-holdback-beta).

**Comment by Amejia481** [comment]
@SoftVision-CiprianMuresan  is the issue only producible on those devices ? Are you able to reproduce on a pixel with Android 11.

**Comment by Amejia481** [comment]
I found the root cause, I put a PR for it https://github.com/mozilla-mobile/fenix/pull/20933

**Comment by amedyne** [comment]
@Amejia481 after validating the fix, we have to uplift the fix to Beta 92.

**Comment by Amejia481** [comment]
👍🏽

**Comment by Amejia481** [comment]
👋🏽 @SoftVision-CiprianMuresan
We put a fix on Beta, could you verify is the issue is fixed there?

**Comment by Amejia481** [comment]
The same fix should be available in `main`

**Comment by Softvision-PatriciuPop** [comment]
We've retested the issue and noticed the notification that the app must be closed for the changes to apply. However, tapping the back button before the app has a chance to close will trigger the same crash and we would like to validate a few more scenarios tomorrow.

**Comment by SoftVision-CiprianMuresan** [comment]
Hey @Amejia481, we've retested the issue again today on more devices (Samsung Galaxy S9 - Android 8 (yesterday), OnePlus 6T - Android 10, OxygenOS 10.3.12; HUAWEI P20 lite, Android 8.0.0, EMUI 8.0.0; Google Pixel 2XL - Android 10) and all of them had the toast message when trying to disable studies. We've used a local Nightly build as well as a local Beta6 build to perform these tests.

There is one thing I've seen on Nightly however, it looks like there is a way to remove individual experiments now and that prompts a dialog box informing the user that the application needs to be closed in order for the changes to apply. 
@Amejia481, My question is: could we do something similar for the `Studies` toggle? As it is now, I think users could be a bit surprised by the notification as there is no mention of needing a restart prior to the toggle.

Attached a screen recording of the dialog for removing an individual experiment:
![dialog menu for removing experiments](https://user-images.githubusercontent.com/18304511/130619037-e12e58e3-eafb-4cf7-9339-c66a6e258368.gif)

**Comment by Amejia481** [comment]
Thank you for testing, yes let's do that 👍🏽

**Comment by Amejia481** [comment]
The dialog fix is already on beta (`releases_v92.0.0`) and fix is coming to nightly https://github.com/mozilla-mobile/fenix/pull/20994

**Comment by SoftVision-CiprianMuresan** [comment]
We've verified the dialog fix on a local Nightly build as well as a local Beta build on the following devices: 
- Samsung Galaxy S9 - Android 8; 
- OnePlus 6T - Android 10, OxygenOS 10.3.12; 
- HUAWEI P20 lite, Android 8.0.0, EMUI 8.0.0; 
- Google Pixel 2XL - Android 10;
- Samsung Galaxy S8, Android 9 (One UI 1.0);

Now there's no way a user could get back into the initial state presented in the original report. 

Ty @Amejia481 for the quick fix! Closing this issue as fixed and marking as verified.
